cmd/abc/abc.go

```diff
@@ -104,9 +107,56 @@ func setLogEnvVars() {
 	}
 }
 
+// checkForUpdates asynchronously checks for updates and prints results to
+// returned buffer.
+// A sync.WaitGroup is returned so main can wait for completion before exiting.
+func checkForUpdates(ctx context.Context) func() {
+	updatesCh := make(chan string, 1)
+
+	go func() {
+		defer close(updatesCh)
+
+		var b bytes.Buffer
+		if err := abcupdater.CheckAppVersion(ctx, &abcupdater.CheckVersionParams{
```

Comment:
Why not have CheckAppVersion return string instead of writing to a writer? Could be simpler.

Reply:
Yeah, I had inherited that design, not quite sure what the reasoning was, my guess was to make it non-blocking so you could run it in a goroutine (since `go func()` doesn't capture the return of `func()`)

Now that its been refactored, string return would probably make more sense